app/mailers/notification_mailer.rb

```diff
@@ -117,7 +117,7 @@ def new_reply(reply, user)
 
     @reply = reply
     @user = user
-
+    return if EmailAddress.all.to_a.collect(&:email).include?(user.email.to_s)
```

Comment:
Could you replace `EmailAddress.all.to_a.collect(&:email)` with `EmailAddress.pluck(:email)` here?